packages/kbn-esql-editor/src/editor_footer/esql_starred_queries_service.tsx

```diff
@@ -110,12 +116,12 @@ export class EsqlStarredQueriesService {
   }
 
   async addStarredQuery(item: Pick<QueryHistoryItem, 'queryString' | 'status'>) {
-    const favoriteItem = {
+    const favoriteItem: { id: string; metadata: StarredQueryMetadata } = {
       id: generateId(),
       metadata: {
         queryString: getTrimmedQuery(item.queryString),
-        timeRan: new Date().toISOString(),
-        status: item.status,
+        createdAt: new Date().toISOString(),
+        status: item.status ?? 'success',
```

Comment:
With types tightening found one potential bug. status is optional in QueryHistoryItem but requried in our modal. 
Is this fine to keep as is this or other suggestions? Or should it be optional in the modal?

Reply:
No this looks great, good catch